typing/typemod.ml

```diff
@@ -3500,9 +3487,20 @@ let type_toplevel_phrase env sig_acc s =
   Typecore.optimise_allocations ();
   (str, sg, to_remove_from_sg, shape, env)
 
-let type_module_alias =
-  type_module_maybe_hold_locks ~alias:Yes_hold_locks true false None
+let type_module_alias env smod =
+  let md, shape, held_locks =
+    type_module_maybe_hold_locks ~alias:true ~hold_locks:true true false
+      None env smod
+  in
+  let locks =
+    match held_locks with
+    | None -> Env.locks_empty
+    | Some (locks, _, _) -> locks
```

Comment:
What are we potentially throwing away here? Not sure what the `Longident.t` represents here in particular

Reply:
We are throwing away `lid` and `loc`, both of which are passed to `env.walk_locks` only used for error printing. This function is only used by `typecore` which doesn't need those info.